app/src/main/java/com/tab/mybadges/ui/badgeList/BadgeListViewModel.kt

```diff
@@ -0,0 +1,12 @@
+package com.tab.mybadges.ui.badgeList
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.ViewModel
+import com.tab.mybadges.MockBadgeList
+import com.tab.mybadges.entities.Badge
+
+class BadgeListViewModel: ViewModel() {
+    fun getBadges(): LiveData<List<Badge>> {
```

Comment:
This is more of a question on why you chose this approach. This is not exactly how I thought of doing this and that's why I'm asking (what I'm trying to say is: you don't need to see this as a request to change or anything like that, I'm just curious and would like to know your thoughts about it)

What I thought was having an exposed live data here and the methods in this class would apply the changes into this exposed livedata. Something like (check the exact code because I'm doing it but not checking if the code is right)

`private val badgesMutableLiveData: MutableLiveData<List<Badge>>
  val badgesLiveData: LiveData<List<Badge>>
        get() = badgesMutableLiveData

  fun getBadges() {
      badgesMutableLiveData.postValue(MockBadgeList.createMockedBadgeListLiveData())
   }`

Just wanted to hear your way of thinking. Maybe the way you done it would give some more flexibility, I don't know..